### ISSUE TITLE
Categorise remaining sources for relicensing purposes.

### ISSUE THREAD

**Issue by OneDeuxTriSeiGo** [pine64/OpenPineBuds#94]
Parent Issue #76 

Group sources based on use to assist re-licensing effort (in comments below).

**Comment by OneDeuxTriSeiGo** [comment]
# Low Priority
## Test Code

Requested License: `Apache-2.0` or `LGPL-3.0-or-greater`.

- [ ] `services/anc_spp_tool/anc_parse_data.h`
- [ ] `services/app_ibrt/inc/app_ibrt_auto_test_cmd_handle.h`
- [ ] `services/app_ibrt/inc/app_ibrt_auto_test.h`
- [ ] `services/app_ibrt/src/app_ibrt_auto_test_cmd_handle.cpp`
- [ ] `services/app_ibrt/src/app_ibrt_auto_test.cpp`
- [ ] `services/auto_test/at_thread.cpp`
- [ ] `services/auto_test/at_thread_user.cpp`
- [ ] `services/auto_test/auto_test.cpp`
- [ ] `services/ble_app/app_tota/app_tota_ble.c`
- [ ] `services/ble_app/app_tota/app_tota_ble.h`
- [ ] `services/ble_profiles/tota/tota_ble.c`
- [ ] `services/ble_profiles/tota/tota_ble.h`
- [ ] `services/ble_profiles/tota/tota_task.c`
- [ ] `services/ble_profiles/tota/tota_task.h`
- [ ] `services/tota/app_spp_tota.cpp`
- [ ] `services/tota/app_spp_tota_general_service.cpp`
- [ ] `services/tota/app_spp_tota_general_service.h`
- [ ] `services/tota/app_tota_anc.cpp`
- [ ] `services/tota/app_tota_anc.h`
- [ ] `services/tota/app_tota_audio_dump.cpp`
- [ ] `services/tota/app_tota_audio_dump.h`
- [ ] `services/tota/app_tota_cmd_handler.cpp`
- [ ] `services/tota/app_tota_conn.cpp`
- [ ] `services/tota/app_tota_conn.h`
- [ ] `services/tota/app_tota.cpp`
- [ ] `services/tota/app_tota_custom.cpp`
- [ ] `services/tota/app_tota_custom.h`
- [ ] `services/tota/app_tota_data_handler.cpp`
- [ ] `services/tota/app_tota_flash_program.cpp`
- [ ] `services/tota/app_tota_flash_program.h`
- [ ] `services/tota/app_tota_general.cpp`
- [ ] `services/tota/app_tota_general.h`
- [ ] `services/tota/app_tota_mic.cpp`
- [ ] `services/tota/app_tota_mic.h`
- [ ] `services/tota/tota_buffer_manager.cpp`
- [ ] `services/tota/tota_buffer_manager.h`
- [ ] `services/tota/tota_stream_data_transfer.cpp`
- [ ] `services/tota/tota_stream_data_transfer.h`
- [ ] `services/tota/tota_test.cpp`
- [ ] `tests/anc_usb/adda_loop_app.c`
- [ ] `tests/anc_usb/adda_loop_app.h`
- [ ] `tests/anc_usb/anc_usb_app.c`
- [ ] `tests/anc_usb/anc_usb_app.h`
- [ ] `tests/anc_usb/dualadc_audio_app.c`
- [ ] `tests/anc_usb/dualadc_audio_app.h`
- [ ] `tests/anc_usb/main.c`
- [ ] `tests/anc_usb/memutils.c`
- [ ] `tests/anc_usb/memutils.h`
- [ ] `tests/anc_usb/mic_key.c`
- [ ] `tests/anc_usb/safe_queue.c`
- [ ] `tests/anc_usb/safe_queue.h`
- [ ] `tests/anc_usb/speech_process.c`
- [ ] `tests/anc_usb/speech_process.h`
- [ ] `tests/anc_usb/usb_audio_app.c`
- [ ] `tests/anc_usb/usb_audio_app.h`
- [ ] `tests/anc_usb/usb_audio_frm_defs.h`
- [ ] `tests/anc_usb/usb_vendor_msg.c`
- [ ] `tests/anc_usb/usb_vendor_msg.h`

## Third Party Apps and Demos

Requested License: `Apache-2.0` or `LGPL-3.0-or-greater`.

- [ ] `thirdparty/demo_lib/demo_lib.c`
- [ ] `thirdparty/demo_lib/demo_lib.h`
- [ ] `thirdparty/userapi/app_thirdparty.cpp`
- [ ] `thirdparty/userapi/app_thirdparty.h`
- [ ] `thirdparty/userapi/demo_app/LibDemo.cpp`
- [ ] `thirdparty/userapi/demo_app/LibDemo.h`
- [ ] `thirdparty/userapi/noise_tracker_app/noise_tracker_callback.c`
- [ ] `thirdparty/userapi/noise_tracker_app/noise_tracker_callback.h`
- [ ] `thirdparty/userapi/noise_tracker_app/NoiseTrackerDemo.cpp`

## Audio Files

Requested License: `CC BY-SA 4.0`

- [ ] `config/_default_cfg_src_/res/SOUND_MUTE.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_ANSWER.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_CHARGE_FINISH.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_CHARGE_PLEASE.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_CONNECTED.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_DIS_CONNECT.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_EIGHT.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_FINDME.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_FIVE.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_FOUR.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_HUNG_UP.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_INCOMING_CALL.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_LANGUAGE_SWITCH.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_NINE.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_ONE.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_OVER.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_PAIR_ENABLE.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_PAIRING_FAIL.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_PAIRING_SUCCESS.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_PAIRING.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_POWER_OFF.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_POWER_ON.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_REFUSE.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_SEVEN.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_SIX.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_THREE.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_TWO.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_WARNING.txt`
- [ ] `config/_default_cfg_src_/res/cn/SOUND_ZERO.txt`
- [ ] `config/_default_cfg_src_/res/en/dudu.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_ALEXA_START.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_ALEXA_STOP.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_ANSWER.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_CHARGE_FINISH.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_CHARGE_PLEASE.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_CONNECTED.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_DIS_CONNECT.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_EIGHT.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_FINDME.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_FIVE.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_FOUR.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_GSOUND_MIC_CLOSE.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_GSOUND_MIC_OPEN.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_GSOUND_NC.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_HUNG_UP.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_INCOMING_CALL.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_LANGUAGE_SWITCH.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_NINE.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_ONE.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_OVER.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_PAIR_ENABLE.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_PAIRING_FAIL.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_PAIRING_SUCCESS.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_PAIRING.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_POWER_OFF.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_POWER_ON.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_REFUSE.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_SEVEN.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_SIX.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_THREE.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_TWO.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_WARNING.txt`
- [ ] `config/_default_cfg_src_/res/en/SOUND_ZERO.txt`
- [ ] `config/_default_cfg_src_/res/eq/EQ_COEF.txt`
- [ ] `config/_default_cfg_src_/res/gs_hw/en_all.txt`
- [ ] `config/_default_cfg_src_/res/ring/SOUND_RING_16000.txt`
- [ ] `config/_default_cfg_src_/res/ring/SOUND_RING_44100.txt`
- [ ] `config/_default_cfg_src_/res/ring/SOUND_RING_48000.txt`
- [ ] `config/_default_cfg_src_/res/ring/SOUND_RING_8000.txt`

## AI Code

Requested License: `Apache-2.0` or `LGPL-3.0-or-greater`.

- [ ] `services/app_ai/inc/app_ai_algorithm.h`
- [ ] `services/app_ai/inc/app_ai_if_config.h`
- [ ] `services/app_ai/inc/app_ai_if_custom_ui.h`
- [ ] `services/app_ai/inc/app_ai_if_gsound.h`
- [ ] `services/app_ai/inc/app_ai_if.h`
- [ ] `services/app_ai/inc/app_ai_if_thirdparty.h`
- [ ] `services/app_ai/inc/app_ai_manager_api.h`
- [ ] `services/app_ai/inc/app_ai_tws.h`
- [ ] `services/app_ai/src/app_ai_algorithm.cpp`
- [ ] `services/app_ai/src/app_ai_if.cpp`
- [ ] `services/app_ai/src/app_ai_if_custom_ui.cpp`
- [ ] `services/app_ai/src/app_ai_if_gsound.cpp`
- [ ] `services/app_ai/src/app_ai_if_thirdparty.cpp`
- [ ] `services/app_ai/src/app_ai_manager_api.cpp`
- [ ] `services/app_ai/src/app_ai_tws.cpp`
- [ ] `services/app_ai/voice_sbc/voice_sbc.cpp`
- [ ] `services/app_ai/voice_sbc/voice_sbc.h`

## Unused

- [ ] `apps/btusbaudio/btusb_audio.c`
- [ ] `apps/btusbaudio/btusb_audio.h`
- [ ] `apps/btusbaudio/usbaudio_thread.c`
- [ ] `apps/btusbaudio/usbaudio_thread.h`
- [ ] `apps/sdmmc/app_sdmmc.cpp`
- [ ] `apps/sdmmc/app_sdmmc.h`
- [ ] `apps/usbaudio/usbaudio_entry.c`
- [ ] `apps/usbhost/app_usbhost.c`
- [ ] `apps/usbhost/app_usbhost.h`
- [ ] `log.txt`
- [ ] `notes.txt`
- [ ] `services/ble_stack/common/api/lePhone_rw_ble_error.txt`
- [ ] `services/nv_section/fpga_section/nvrecord_ble.c`
- [ ] `services/nv_section/fpga_section/nvrecord_bt.c`
- [ ] `services/nv_section/fpga_section/nvrecord_env.c`
- [ ] `services/nv_section/fpga_section/nvrecord_externsion.c`
- [ ] `utils/kfifo/kfifo.c`

**Comment by nar001** [comment]
So curious, how much is left? Has there been any news about this?

**Comment by shymega** [comment]
No, we're still working on it. It doesn't look like BES will co-operate with what they need to do, but I think we should push on this.

**Comment by nar001** [comment]
I imagine it would be a lot slower, is there a chance BES will wake up and help, or will you have to basically do all of it yourself? Is it about rewriting the code to be clean-room?

**Comment by shymega** [comment]
There's a few binary blobs which we have no idea what they do, for ANC and the RTOS. I believe BES were going to provide us with FreeRTOS that links in with the rest, but I don't know what happened there.

Personally, I'm working to use a LLM to transform the Makefiles into CMake-style build system. Doing it manually has been painful. That's one of the prerequisites for further work on the project.

**Comment by nar001** [comment]
Fair enough, I hope they'll help, if only so that's done! I imagine not a lot of people are working on that either, I hope it's not too time consuming!

**Comment by Ralim** [comment]
Where communication with BES ended up at is (afaik) :

1. CEO was all for the re-license, and having the software team rebuild the blobs for FreeRToS.
2. CEO then asked their software team lead (Not sure on exact name here), to provide the blobs and something to officialise the re-license.
3. That person then did nothing.
4. TL re-met with the CEO a few months later, they confirmed they they were happy with it and that they would follow up.
5. CEO is/was sick so couldn't meet the team in person to talk about it
6. Repeat steps 1-5 a few times, more than once per year since we first looked at this chip _years ago_. We were sitting at step 2 right about PBP announcement/manufacturing started.

At this point, I'm starting to think the software team is just avoiding doing anything to help us at all. TL keeps to work on it every time he can meet up with the BES CEO, but I haven't seen forward progress in a long time now 😢

**Comment by WebWire-NL** [comment]
Concrete input for this: the prebuilt `*.a` archives are compiled with `-g` and name their own sources, which gives a categorised list of what is still closed - 498 files, bucketed by directory, from DWARF rather than by hand:

| files | directory |
|---|---:|
| 174 | `services/multimedia/audio/codec` |
| 130 | `services/multimedia/speech/src` |
| 34 | `services/ble_stack/hl/src` |
| 27 | `services/multimedia/audio/process` |
| 14 | `thirdparty/audio_codec_lib/ldac/src` |
| 9 | `thirdparty/audio_codec_lib/liblhdc-dec/dec_src` |
| 7 | `services/ble_stack/ke/src` |
| 6 | `services/bt_profiles_enhanced/protocols/btm` |
| 5 | `services/bt_profiles_enhanced/protocols/obex` |
| 4 | `services/ble_stack/hci/src` |
| 4 | `services/bt_profiles_enhanced/protocols/hshf` |
| 3 | `services/bt_profiles_enhanced/protocols/besaud`, `.../map`, `.../misc`, `.../pbap` |

(the tail is smaller groups; the full per-library breakdown including which object names which source is in the report below)

- prose report, method and per-library table: https://github.com/WebWire-NL/openpinebuds-integration/blob/main/docs/closed-blobs.md
- per-function records with the vendor source file and line each declaration came from: https://github.com/WebWire-NL/openpinebuds-integration/blob/main/data/headerless-api.json
- summary in the tree: pine64/OpenPineBuds#126

Three categorisation notes that matter for relicensing:
- **Third-party, not BES's to relicense**: Fraunhofer FDK-AAC (`services/multimedia/audio/codec/fdkaac_codec` - evidenced by 1,282 name-mangled C++ symbols and the `aacDecoder_*` API), Savitech LHDC (`liblhdc-dec`) and Sony LDAC. They need a separate bucket from "ask BES".
- **"Still closed" here means "named by the blobs, absent from this tree"** - some of these files exist in other public BES SDK trees in different chip generations, which is a different category again: https://github.com/WebWire-NL/openpinebuds-integration/blob/main/docs/prior-art.md
- Some of the list is codec/DSP *interface* material only: 228 of the recovered declarations are already covered by headers that ship in this tree.

Happy to emit the whole list in whatever per-file format #93 uses (path, declared licence, requested licence, reconstruction complete/partial). Metadata only - no vendor source, no decompiled code. Credit: @WebWire-NL.
